### DISCUSSION TITLE
Is it possible to disable Fast Refresh in React?

### DISCUSSION THREAD

**Discussion by Chevindu** [web-infra-dev/rsbuild#3733] [Q&A] [answered]
I see on the docs that for React projects, Rsbuild uses React's official Fast Refresh capability to perform component hot updates. Is there any way to turn this off and use React's old HMR? With webpack, I could use `FAST_REFRESH=false` in env variables. 

Context: I have a very old react project that I'd like to try and get rsbuild integrated. It has Class components, so the fast refresh doesn't really work during development.

**Comment by chenjiahan** [comment, marked as answer]
We will add a new option in the next release: https://github.com/web-infra-dev/rsbuild/pull/3738
